website/blog/2024-03-21-using-your-private-git-repository-context-for-completion/index.md

```diff
@@ -0,0 +1,146 @@
+---
+authors: [icycodes]
+tags: [deployment, repository context]
+---
+
+# Utilizing Your Private Git Repository Context for Enhanced Code Completion
+
+A few months back, we published a blog [Repository context for LLM assisted code completion](https://tabby.tabbyml.com/blog/2023/10/16/repository-context-for-code-completion), introducing the `Repository Context` feature in Tabby. 
+
+In this blog, I will guide you through the process of setting up a Tabby server configured with a private Git repositories context.
+
+## Generating a Personal Access Token for Your Private Git Repository
+
+In order to provide the Tabby server with access to your private Git repositories, it is essential to create a Personal Access Token (PAT) specific to your Git provider. The following steps outline the process with GitHub as a reference:
+
+1. Visit [GitHub Personal Access Tokens Settings](https://github.com/settings/tokens?type=beta) and select `Generate new token`.
+  ![GitHub PAT Generate New Token](./github-pat-generate-new-token.png)
+2. Enter the `Token name`, specify an `Expiration` date, an optional `Description`, and select the repositories you wish to grant access to.
+  ![GitHub PAT Filling Info](./github-pat-filling-info.png)
+3. Within the `Permissions` section, ensure that `Contents` is configured for `Read-only` access.
+  ![GitHub PAT Contents Access](./github-pat-contents-access.png)
+4. Click `Generate token` to generate the new PAT. Remember to make a copy of the PAT before closing the webpage.
+  ![GitHub PAT Generate Token](./github-pat-generate-token.png)
+
+For additional information, please consult the documentation on [Managing your personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).
+
+**Note**: For users of GitLab, guidance on creating a personal access token can be found in the documentation [Personal access tokens - GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#create-a-personal-access-token).
+
+## Creating a Configuration File for Tabby Server
```

Comment:
use `--webserver` based configuration (as git repository is part of community tier)